lib/controllers/issueish-detail-controller.js

```diff
@@ -44,14 +71,124 @@ export class BareIssueishDetailController extends React.Component {
       return <div>Issue/PR #{this.props.issueishNumber} not found</div>; // TODO: no PRs
     }
 
+    this.checkoutOp = this.nextCheckoutOp();
+
     return (
       <IssueishDetailView
         repository={repository}
         issueish={repository.issueish}
+        checkoutOp={this.checkoutOp}
         switchToIssueish={this.props.switchToIssueish}
       />
     );
   }
+
+  nextCheckoutOp() {
+    const {repository} = this.props;
+    const {issueish} = repository;
+
+    if (issueish.__typename !== 'PullRequest') {
+      return this.checkoutOp.disable(checkoutStates.HIDDEN, 'Cannot check out an issue');
+    }
+
+    if (this.props.isAbsent) {
```

Comment:
would it maybe make sense to use some sort of enum prop here so you're not passing allll the props?  And then you could use a switch statement? (I don't feel super strongly, just a suggestion)

Reply:
Hmm you're right that it's awkward to query a bunch of `isXyz` props here. There's an enum-ish pattern that I use sometimes that might be nice for repository state branching like this:

```js
return this.props.repository.inState({
  absent: () => this.checkoutOp.disable(checkoutStates.HIDDEN, 'No repository found'),
  loading: () => this.checkoutOp.disable(checkoutStates.DISABLED, 'Loading'),
  // ...
  default: () => {},
});
```

Another alternative would be to pull more of this logic into the active Repository state:

```js
return this.props.repository.checkoutOpEnablement(this.checkoutOp);

// lib/models/repository-states/absent.js
export default class Absent extends State {
  checkoutOpEnablement(checkoutOp) {
    return checkoutOp.disable(checkoutStates.DISABLED, 'No repository found');
  }
}

// lib/models/repository-states/loading.js
export default class Loading extends State {
  checkoutOpEnablement(checkoutOp) {
    return checkoutOp.disable(checkoutStates.DISABLED, 'Loading');
  }
}
```

There are a few places that we do this already, though, so I think I'd rather address it in a code-cleanup PR rather than here, to be sure that we can come up with something that works nicely across the codebase 😄